pulp_2_tests/tests/rpm/api_v2/test_modularity.py

```diff
@@ -0,0 +1,65 @@
+# coding=utf-8
+"""Tests that perform actions over RPM modular repositories."""
+import os
+import unittest
+
+from packaging.version import Version
+from pulp_smash import api, cli, config
+from pulp_smash.pulp2.constants import REPOSITORY_PATH
+from pulp_smash.pulp2.utils import publish_repo, sync_repo
+
+from pulp_2_tests.constants import RPM_WITH_MODULES_FEED_URL
+from pulp_2_tests.tests.rpm.api_v2.utils import (
+    gen_distributor,
+    gen_repo,
+)
+from pulp_2_tests.tests.rpm.utils import set_up_module as setUpModule  # pylint:disable=unused-import
+
+
+class ManageModularContentTestCase(unittest.TestCase):
+    """Manage modular content tests cases.
+
+    This test targets the following issue:
+
+    `Pulp Smash #1122 <https://github.com/PulpQE/pulp-smash/issues/1122>`_
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class wide variables."""
+        cls.cfg = config.get_config()
+        if cls.cfg.pulp_version < Version('2.17'):
+            raise unittest.SkipTest('This test requires Pulp 2.17 or newer.')
+        cls.client = api.Client(cls.cfg, api.json_handler)
+
+    def test_sync_publish_repo(self):
+        """Test sync and publish modular RPM repository."""
+        body = gen_repo()
+        body['importer_config']['feed'] = RPM_WITH_MODULES_FEED_URL
+        body['distributors'] = [gen_distributor()]
+        repo = self.client.post(REPOSITORY_PATH, body)
+        self.addCleanup(self.client.delete, repo['_href'])
+        sync_repo(self.cfg, repo)
+        repo = self.client.get(repo['_href'], params={'details': True})
+
+        # Assert that `modulemd` and `modulemd_defaults` are present on the
+        # repository.
+        self.assertIsNotNone(repo['content_unit_counts']['modulemd'])
+        self.assertIsNotNone(repo['content_unit_counts']['modulemd_defaults'])
+
+        publish_repo(self.cfg, repo)
+
+        with self.subTest(comment='verfiy that a *modules.yaml.gz exists'):
+            modules_yaml_gz = self.find_modules_yaml_gz(repo)
+            self.assertEqual(len(modules_yaml_gz), 1, modules_yaml_gz)
+
+    def find_modules_yaml_gz(self, repo):
+        """Return a list of ``modules.yaml.gz`` in ``/var/lib/published/yum/http/repos``."""
```

Comment:
This docstring is misleading. This function doesn't look in `/var/lib/published/yum/http/repos`. It looks in `/var/lib/published/yum/http/repos/{relative_ur}/repodata`. It also does not look for files named `modules.yaml.gz`. It looks for files named `*modules.yaml.gz`. Maybe this could read:

    """Find ``*modules.yaml.gz`` files in the given repo's ``repodata`` directory."""